glib-macros/src/gflags_attribute.rs

```diff
@@ -0,0 +1,184 @@
+
+use heck::{CamelCase, KebabCase, SnakeCase};
+use proc_macro2::TokenStream;
+use proc_macro_error::abort_call_site;
+use quote::{format_ident, quote, quote_spanned};
+use syn::{
+    punctuated::Punctuated, spanned::Spanned, token::Comma, Attribute, Data, DeriveInput, Ident,
+    LitStr, Variant,
+};
+
+use crate::utils::{find_attribute_meta, find_nested_meta, parse_item_attributes, ItemAttribute};
+
+// Flag is not registered if it has the #[gflags(skip)] meta
+fn attribute_has_skip(attrs: &[Attribute]) -> bool {
+    let meta = find_attribute_meta(attrs, "gflags").unwrap();
+
+    match meta {
+        None => false,
+        Some(meta) => find_nested_meta(&meta, "skip").is_some(),
+    }
+}
+
+// Generate gobject_sys::GFlagsValue structs mapping the enum such as:
+//     gobject_sys::GFlagsValue {
+//         value: MyFlags::A.bits(),
+//         value_name: "The Name\0" as *const _ as *const _,
+//         value_nick: "nick\0" as *const _ as *const _,
+//     },
+fn gen_gflags_values(
+    enum_name: &Ident,
+    enum_variants: &Punctuated<Variant, Comma>,
+) -> (TokenStream, usize) {
+    // start at one as GFlagsValue array is null-terminated
+    let mut n = 1;
+    let recurse = enum_variants.iter().filter(|v| { !attribute_has_skip(&v.attrs) } ).map(|v| {
+        let name = &v.ident;
+        let mut value_name = name.to_string().to_camel_case();
+        let mut value_nick = name.to_string().to_kebab_case();
+
+        let attrs = parse_item_attributes("gflags", &v.attrs);
+        let attrs = match attrs {
+            Ok(attrs) => attrs,
+            Err(e) => abort_call_site!(
+                "{}: gflags enum supports only the following optional attributes: #[gflags(name = \"The Name\", nick = \"the-nick\")] or #[gflags(skip)]",
+                e
+            ),
+        };
+
+        attrs.into_iter().for_each(|attr|
+            match attr {
+                ItemAttribute::Name(n) => value_name = n,
+                ItemAttribute::Nick(n) => value_nick = n,
+            }
+        );
+
+        let value_name = format!("{}\0", value_name);
+        let value_nick = format!("{}\0", value_nick);
+
+        n += 1;
+        quote_spanned! {v.span()=>
+            gobject_sys::GFlagsValue {
+                value: #enum_name::#name.bits(),
+                value_name: #value_name as *const _ as *const _,
+                value_nick: #value_nick as *const _ as *const _,
+            },
+        }
+    });
+    (
+        quote! {
+            #(#recurse)*
+        },
+        n,
+    )
+}
+
+fn gen_bitflags(enum_name: &Ident, enum_variants: &Punctuated<Variant, Comma>) -> TokenStream {
+    let recurse = enum_variants.iter().map(|v| {
+        let name = &v.ident;
+        let disc = v.discriminant.as_ref().expect("missing discriminant");
+        let value = &disc.1;
+
+        quote_spanned! {v.span()=>
+            const #name = #value;
+        }
+    });
+
+    quote! {
+        bitflags::bitflags! {
```

Comment:
Maybe should re-export it hidden from the `glib` crate and explicitly use that from here?